libs/blocks/quiz-entry/quiz-entry.js

```diff
@@ -73,127 +116,150 @@ const App = ({
-        if (currentFlow && currentFlow.length) {
-          setSelectedQuestion(quizLists.questions[currentFlow] || []);
+        if (Object.keys(mlmap.mlDetails).length > 0) {
+          setHasMLData(true);
+        } else {
+          setHasMLData(false);
         }
       }
     })();
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [quizState, quizLists]);
+  }, [selectedQuestion]);
 
-  useEffect(() => {
-    if (countSelectedCards > 0) {
-      setCardsUsed(true);
-    } else {
-      setCardsUsed(false);
-    }
-  }, [countSelectedCards]);
+  const continueQuiz = async () => {
+    let selections = {};
+    let userFlow = [];
+    if (mlInputUsed) {
+      const { mlDetails, mlValues } = mlData;
+      const mlFieldText = document.querySelector('#ml-field-input').value;
+      const fiResults = await getMLResults(mlDetails.endpoint, mlDetails['api-key'], mlDetails.threshold || defaultThreshold, mlFieldText, fiCodeCount, mlValues);
+      const { filtered } = fiResults;
 
-  // let minSelections = 0;
-  let maxSelections = 10;
+      filtered.forEach((item) => {
+        selections[item.ficode] = true;
+      });
 
-  if (selectedQuestion) {
-    // minSelections = +selectedQuestion['min-selections'];
-    maxSelections = +selectedQuestion['max-selections'];
-  }
+      if (debug) {
+        // eslint-disable-next-line no-console
+        console.log('all', fiResults.data);
+        // eslint-disable-next-line no-console
+        console.log('filtered', filtered);
+      }
+    }
 
-  const getStringValue = (propName) => {
-    if (!selectedQuestion?.questions) return '';
-    const question = quizLists.strings[selectedQuestion.questions];
-    return question?.[propName] || '';
-  };
+    if (cardsUsed) {
+      userFlow = quizState.userFlow;
+      selections = selectedCards;
+      if (questionCount.current < maxQuestions) {
+        setSelectedCards({});
+        setSelectedQuestion(null);
+      }
+    }
 
-  const getOptionsValue = (optionsType, prop) => {
-    const optionItem = quizData.strings[selectedQuestion.questions].data.find(
-      (item) => item.options === optionsType,
-    );
-    return optionItem && optionItem[prop] ? optionItem[prop] : '';
-  };
+    if (Object.keys(selections).length > 0) {
+      const { nextFlow } = handleNext(
+        questionData,
+        selectedQuestion,
+        selections,
+        userFlow,
+      );
+      const { nextSelections } = handleSelections(
+        quizState.userSelection,
+        selectedQuestion,
+        selections,
+      );
 
-  const onMLInput = (event) => {
-    const inputValue = event.target.value;
-    if (inputValue.length > 0) {
-      if (!mlInputUsed) setMLInputUsed(true);
-    } else {
-      setMLInputUsed(false);
-    }
-  };
+      const currentQuizState = {
+        userFlow: nextFlow,
+        userSelection: nextSelections,
+      };
+      localStorage.setItem('stored-quiz-state', JSON.stringify(currentQuizState));
```

Comment:
do we need to consider locale like quiz-key?

Reply:
No. First because the quiz block is looking for this locale storage key specifically and it doesn't benefit from locale like quiz-key does, finally, this is not displayed to the user in any way - quiz-key is seen in the query string. Second, they don't have any intentions of localizing this unless it wins in their testing. If it does win, see point one - no need to add locale to this.